### PR TITLE
 Bug fixes, many improvements, and updates to match offical repository

### PR DESCRIPTION
**Pull request by rlronan**
## Fixes:
- Moved model weight loading into the training loop in train.py, so it works properly. The model needs to be initialized for the weights to be loaded.
- Fixed code for Discriminator weight loading. 
- Divided DownSamplingBlock output by two so that it is an average of the two sub-outputs, rather than a sum.

## Updates:
- Added Tensorboard metric logging for every step, rather than just every epoch.
- Added Visualizations at every 500 steps if the epoch is longer than 500 steps.
- Changed image resizing in training_routine.py to use 'nearest' method since the resizing is by a factor of 1/2. 
- create_dataset function now also tries to find .png files if no .jpg files are found. 
- center_crop_image function now performs random crops, but is still called 'center_crop_images'
- Replaced BatchNormalization with LayerNormalization everywhere, as it is more appropriate to use with small batch sizes. 
- Updated tensorboard logging to take steps rather than epochs. 

## Updates to match official implementation
- Replaced reconstruction loss MAE function with perception loss as per the official repository implementation, and because MAE is not a good image distance metric.  
- Included a forked version of https://github.com/alexlee-gk/lpips-tensorflow  which i updated for TF2.x to perform the perception loss.
- In the discriminator, added a spectral norm wrapper to every conv layer, and updated leakyrelu rate to 0.2
- In the discriminator, added SLE blocks as per the official repository; this should improve the gradient flow in the discriminator. 
- In the generator, included the code for upsampling blocks that match the official repository (under UpsamplingBlock2). These have a small amount of Gaussian noise added, and are twice as many layers as the current UpsamplingBlock. In the official repository, both types of upsampling blocks are interspersed.  
- Turned bias off for every conv layer in the discriminator and generator.
- Replaced leakyrelu with silu (aka swift) activation function in SLE blocks.

### CODE REVIEW

**Comment by gaborvecsei** [comment]
First of all, thanks for the contribution.

I tried a training with these "fixes" but I did not get the expected results. Can you tell more what hyper parameters did you use and what are the results of your own training?

## Loss curves

### Your branch

image
image

## "Whole Reconstruction" images

Randomly picked from the end of the training (1 day of training with a V100) - but this is how it looks like after each step/epoch.

image

image

**Comment by gaborvecsei** [comment]
@rlronan Any update on your PR or the results I got?

**Comment by gaborvecsei** [comment]
Closing because of inactivity
